lightning/src/chain/channelmonitor.rs

```diff
@@ -643,6 +643,7 @@ struct IrrevocablyResolvedHTLC {
 	/// was not present in the confirmed commitment transaction), HTLC-Success, or HTLC-Timeout
 	/// transaction.
 	resolving_txid: Option<Txid>, // Added as optional, but always filled in, in 0.0.110
+	resolving_tx: Option<Transaction>,
```

Comment:
hm, well, the alternative is consistently enforcing the resoving_txid == resolving_tx.map(|tx| tx.txid()) invariant, so I think removing it might make things less confusing down the line, but doesn't have to happen right now.